core/src/main/scala/org/apache/spark/storage/BlockManager.scala

```diff
@@ -650,6 +658,23 @@ private[spark] class BlockManager(
       blockId: BlockId,
       level: StorageLevel,
       classTag: ClassTag[_]): StreamCallbackWithID = {
+
+    if (decommissioner.isDefined) {
+       throw new BlockSavedOnDecommissionedBlockManagerException(blockId)
+    }
+
+    if (blockId.isShuffle || blockId.isInternalShuffle) {
+      logInfo(s"Putting shuffle block ${blockId}")
+      try {
+        return migratableResolver.putShuffleBlockAsStream(blockId, serializerManager)
+      } catch {
+        case e: ClassCastException => throw new Exception(
```

Comment:
`Exception` -> `SparkException`?

Reply:
BTW, I'm wondering if we have a test case for this code path?

Reply:
Good point, I don't. I'll add a unit test for this.